### PR TITLE
Automate vNIC tests

### PR DESCRIPTION
**Pull request by yossisegev**
Automate vNIC tests:
- Verify default NIC model type is VirtIO.
- Verify unsupported NIC model type cannot be applied.

### CODE REVIEW

**Comment by yossisegev** [comment]
@dankenigsberg @myakove 
Please review

File: tests/utils.go [hunk 1]
```diff
@@ -1772,6 +1772,14 @@ func NewRandomVMIWithe1000NetworkInterface() *v1.VirtualMachineInstance {
 	return vmi
 }
 
+func NewRandomVMIWithInvalidNetworkInterface() *v1.VirtualMachineInstance {
```
Comment (myakove) [inline tests/utils.go]: I guess this function can be in the test file and not in utils.

**Comment by SchSeba** [comment]
@yossisegev also please run `make generate` and update the PR to make travis happy

File: tests/utils.go [hunk 2]
```diff
@@ -63,12 +63,12 @@ import (
 	"k8s.io/client-go/tools/remotecommand"
 
 	cdiv1 "kubevirt.io/containerized-data-importer/pkg/apis/datavolumecontroller/v1alpha1"
-	"kubevirt.io/kubevirt/pkg/api/v1"
+	v1 "kubevirt.io/kubevirt/pkg/api/v1"
```
Comment (cynepco3hahue) [inline tests/utils.go]: redundant

File: tests/utils.go [hunk 3]
```diff
@@ -39,7 +39,7 @@ import (
 	"time"
 
 	ghodssyaml "github.com/ghodss/yaml"
-	"github.com/google/goexpect"
+	expect "github.com/google/goexpect"
```
Comment (cynepco3hahue) [inline tests/utils.go]: use explicit names for packages only when you have a conflict with another package

Comment (yossisegev) [reply tests/utils.go]: VSCode entered these explicit package names automatically once I saved each of these files, so I couldn't even remove them (unless I stopped using vscode).

File: tests/utils.go (hunk 2 above)
Comment (yossisegev) [reply tests/utils.go]: VSCode entered these explicit package names automatically once I saved each of these files, so I couldn't even remove them (unless I stopped using vscode).

File: tests/utils.go (hunk 1 above)
Comment (yossisegev) [reply tests/utils.go]: I put this function here as a "sister" function to NewRandomVMIWithe1000NetworkInterface(), which is also implemented here although it is only used in vmi_networking_test.go.

Comment (SchSeba) [reply tests/utils.go]: @yossisegev you can do something like this directly in the test.
```golang
vmi := NewRandomVMIWithe1000NetworkInterface()
vmi.Spec.Domain.Devices.Interfaces[0].Model = "gibberish"
_, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
Expect(err).To(HaveOccurred())
```

Comment (yossisegev) [reply tests/utils.go]: @SchSeba , I'll go with with moving this function to the test file itself. Using NewRandomVMIWithe1000NetworkInterface() seems out of context of this specific test.

Comment (SchSeba) [reply tests/utils.go]: sounds good

File: tests/utils.go (hunk 3 above)
Comment (yossisegev) [reply tests/utils.go]: Resolved this issue in VSCode with the help of Meni.

File: tests/utils.go (hunk 2 above)
Comment (yossisegev) [reply tests/utils.go]: Resolved this issue in VSCode with the help of Meni.

**Comment by yossisegev** [comment]
CI please test

**Pull request merged**